tracing/tracing.go

```diff
@@ -5,46 +5,37 @@ import (
 	"io"
 	"io/ioutil"
 	"os"
-	"strconv"
 
 	jaegercfg "github.com/uber/jaeger-client-go/config"
 )
 
-// New registers Jaeger as the OpenTracing implementation.
-// If jaegerAgentHost is an empty string, tracing is disabled.
-func New(agentHost, serviceName, samplerType string, samplerParam float64) io.Closer {
-	if agentHost != "" {
-		cfg := jaegercfg.Configuration{
-			Sampler: &jaegercfg.SamplerConfig{
-				SamplingServerURL: fmt.Sprintf("http://%s:5778/sampling", agentHost),
-				Type:              samplerType,
-				Param:             samplerParam,
-			},
-			Reporter: &jaegercfg.ReporterConfig{
-				LocalAgentHostPort: fmt.Sprintf("%s:6831", agentHost),
-			},
-		}
-
-		closer, err := cfg.InitGlobalTracer(serviceName)
-		if err != nil {
-			fmt.Printf("Could not initialize jaeger tracer: %s\n", err.Error())
-			os.Exit(1)
-		}
-		return closer
+// InstallJaeger registers Jaeger as the OpenTracing implementation.
+func InstallJaeger(serviceName string, cfg *jaegercfg.Configuration) io.Closer {
```

Comment:
As this is an exported function, I am wondering are we sure we do not use this, and if its only used here can we just not export it?

Reply:
Yeah YAGNI I guess, we don't use it, I was mostly trying to make up for removing the `New` method.